### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by lindakung**
I was able to create a virtual that filtered out the unique challenges of all the challenges a user has completed. But I've only been able to get the unique ID and not actually grab/populate that information yet.

### CODE REVIEW

**Comment by seemaullal** [comment]
This looks great- it looks like the profile page and scoring are coming along. I love your solution for getting unique exercise Ids- it's awesome!

Before you merge, I had this idea for getting the whole exercise instead of just the Id

For this part -
    this.exercises.forEach(function(exercise) {
        if(!unique[exercise.exerciseID]) {
            ids.push(exercise.exerciseID);
            unique[exercise.exerciseID] = true;
        }
    })
        return ids;
        console.log('uniqueIDs', ids);
})

Try ids.push(exercise) instead of exercise.exerciseID to get the whole exercise. 

Other than that potential change, this looks great and ready to merge after that

File: server/app/configure/authentication/index.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ module.exports = function (app) {
     // logged in already.
     app.get('/session', function (req, res) {
         if (req.user) {
-            res.send({ user: _.omit(req.user.toJSON(), ['salt', 'password']) });
+            res.send({ user: _.omit(req.user.toJSON({virtuals:true}), ['salt', 'password']) });
```
Comment (ghost) [inline server/app/configure/authentication/index.js:51]: WHAT IS THIS!?

**Comment by lindakung** [comment]
@seemaullal i see your point and i tried it but exercises in the user model is a reference so it will only return an id. i think we have to populate the exercises...not positive how yet..

**Comment by seemaullal** [comment]
Ohhhh ok yeah that makes sense . I think when you call the virtual just add .populate after 

So the virtual is the same but calling it is when you populate. In the route for example. 

The virtual shouldn't really be called in angular. It should be more like
Angular-> requests virtual info from backend route -> backend route gets the virtual value and populates (so AJAX requests basically) 

Idk if that made sense. Probably not.

**Comment by seemaullal** [comment]
Looks like you solved this so you should be fine merging. :sheep: :shipit:

**Pull request merged**